File: java/src/test/java/problems101_200/BestTimeToBuyAndSellStockTest.java
package problems101_200;

import org.junit.Test;
import sample.SampledTest;

import static com.google.common.truth.Truth.assertWithMessage;

public class BestTimeToBuyAndSellStockTest extends SampledTest {
    @Test
    public void test() {
        for (BestTimeToBuyAndSellStock solution : BestTimeToBuyAndSellStock.values()) {
            for (BestTimeToBuyAndSellStockSample sample
                : getSamples(BestTimeToBuyAndSellStockSample[].class)) {
                assertWithMessage(sample.getMessage())
                    .that(solution.maxProfit(sample.input.prices))
                    .isEqualTo(sample.output);
            }
        }
    }
}

File: kotlin/src/test/kotlin/problems101_200/BestTimeToBuyAndSellStockTest.kt
package problems101_200

import com.google.common.truth.Truth.assertWithMessage
import sample.SampledTest
import sample.component1
import sample.component2
import sample.component3
import sample.getSamples
import kotlin.test.Test

class BestTimeToBuyAndSellStockTest : SampledTest() {
    @Test
    fun test() =
        BestTimeToBuyAndSellStock.entries.forEach { solution ->
            getSamples<Array<BestTimeToBuyAndSellStockSample>>()
                .forEach { (input, output, message) ->
                    assertWithMessage(message)
                        .that(solution.maxProfit(input.prices))
                        .isEqualTo(output)
                }
        }
}
